addons/mail/static/src/models/suggested_recipient_info/suggested_recipient_info.js

```diff
@@ -37,6 +37,13 @@ function factory(dependencies) {
             return this.partner && this.partner.nameOrDisplayName || this.name;
         }
 
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeLang() {
+            return this.partner && this.partner.lang || this.lang;
+        }
```

Comment:
Better not make this adjusting compute, I don't think you actually need the value once the partner is created (and if you do, you can access it through the partner anyway).

Just keep the lang of the suggested recipient intact, which is the lang originally given by the python for this suggestion.